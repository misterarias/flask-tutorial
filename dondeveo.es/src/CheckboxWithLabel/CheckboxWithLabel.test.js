'use strict'

import React from 'react';
import CheckboxWithLabel from './CheckboxWithLabel';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() })

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <CheckboxWithLabel
      labelOn="ON" labelOff="OFF"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('CheckboxWithLabel changes the text after click', () => {
  // Render a checkbox with label in the document
  const checkbox = shallow(
    <CheckboxWithLabel labelOn="On" labelOff="Off" />
  );

  expect(checkbox.text()).toEqual('Off');

  checkbox.find('input').simulate('change');

  expect(checkbox.text()).toEqual('On');
});
