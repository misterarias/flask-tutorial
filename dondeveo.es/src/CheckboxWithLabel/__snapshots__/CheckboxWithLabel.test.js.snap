// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  OFF
</label>
`;
